src/snowflake/snowpark/session.py

```diff
@@ -917,10 +917,25 @@ def table_function(
         func_expr = _create_table_function_expression(
             func_name, *func_arguments, **func_named_arguments
         )
-        d = DataFrame(
-            self,
-            TableFunctionRelation(func_expr),
-        )
+
+        from snowflake.snowpark import context
+
+        if context._use_sql_simplifier:
+            d = DataFrame(
+                self,
+                SelectStatement(
+                    from_=SelectSnowflakePlan(
+                        snowflake_plan=TableFunctionRelation(func_expr),
+                        analyzer=self._analyzer,
+                    ),
```

Comment:
An alternative way is to create a new class like `SelectableTableFunction` that extends `Selectable`.

Reply:
Let me try to implement it this way because it can be useful when integrating with work I did last sprint on `select` with table function